cypress/utils/plugins/dashboards-assistant/commands.js

```diff
@@ -0,0 +1,140 @@
+  //     );
+  //     /**
+  //      * wait for 2s
+  //      */
+  //     cy.wait(2000);
+  //   } else {
+  //     throw new Error(resp);
+  //   }
+  // });
+  const nodesMap = {};
+  FlowTemplateJSON.workflows.provision.nodes.forEach((node) => {
+    nodesMap[node.type] = nodesMap[node.type] || [];
+    nodesMap[node.type].push(node);
+  });
+  cy.request(
+    'POST',
+    `${BACKEND_BASE_PATH}${ML_COMMONS_API.CREATE_CONNECTOR}`,
+    nodesMap.create_connector[0].user_inputs
+  )
+    .then((resp) => {
+      agentParameters.connectorId = resp.body.connector_id;
+      return cy.request(
+        'POST',
+        `${BACKEND_BASE_PATH}${ML_COMMONS_API.CREATE_MODEL}?deploy=true`,
+        {
+          ...nodesMap.register_remote_model[0].user_inputs,
+          connector_id: agentParameters.connectorId,
+          function_name: 'remote',
+        }
+      );
+    })
+    .then((resp) => {
+      agentParameters.modelId = resp.body.model_id;
+      return cy.request(
+        'POST',
+        `${BACKEND_BASE_PATH}${ML_COMMONS_API.CREATE_AGENT}`,
+        {
+          ...nodesMap.register_agent[0].user_inputs,
+          llm: {
+            parameters:
+              nodesMap.register_agent[0].user_inputs['llm.parameters'],
+            model_id: agentParameters.modelId,
+          },
+          tools: [nodesMap.create_tool[0]].map((item) => item.user_inputs),
+        }
+      );
+    })
+    .then((resp) => {
+      agentParameters.conversationalAgentId = resp.body.agent_id;
+      return cy.request(
+        'POST',
+        `${BACKEND_BASE_PATH}${ML_COMMONS_API.CREATE_AGENT}`,
+        {
+          ...nodesMap.register_agent[1].user_inputs,
+          tools: [
+            {
+              ...nodesMap.create_tool[1].user_inputs,
+              parameters: {
+                ...nodesMap.create_tool[1].user_inputs.parameters,
+                agent_id: agentParameters.conversationalAgentId,
+              },
+            },
+            {
+              ...nodesMap.create_tool[2].user_inputs,
+              parameters: {
+                ...nodesMap.create_tool[2].user_inputs.parameters,
+                model_id: agentParameters.modelId,
+              },
+            },
+          ],
+        }
+      );
+    })
+    .then((resp) => {
+      agentParameters.rootAgentId = resp.body.agent_id;
+    });
+});
+
+Cypress.Commands.add('cleanRootAgent', () => {
+  return;
+  /**
+   * TODO wait for flow framework to be built into snapshot.
+   */
+  // cy.request(
+  //   'POST',
+  //   `${BACKEND_BASE_PATH}${FLOW_FRAMEWORK_API.CREATE_FLOW_TEMPLATE}/${usingWorkflowId}/_deprovision`
+  // );
+  // /**
+  //  * wait for 2s
+  //  */
+  // cy.wait(2000);
+});
+
+Cypress.Commands.add('startDummyServer', () => {
+  // Not a good practice to start a server inside Cypress https://docs.cypress.io/guides/references/best-practices#Web-Servers
+  // But in out case, we need to reuse release e2e template and let's make it a tradeoff.
```

Comment:
Shall we add this `startDummyServer` in the `package.json` or workflow files manually? Just wondering if there are some ports conflicts  after call this command for many times.

Reply:
> add `startDummyServer ` to workflow files manually
We can not because by leveraging the release E2E tremplate, the test-cmd will be passed to Cypress-test step and that step won't accept `&` operators in its input.

> there are some ports conflicts after call this command for many times
yes it will, we've add stopDummyServer in our test command and that should address the ports conflict issue.

Reply:
Can we use like `command-prefix` in the `cypress-io/github-action@v2`? The `nohup` may not work in the windows. Another concern was not sure if the `stopDummyServer` is a sync method. Even it's a sync method, it will close dummy server for other tests, if we run tests in parallel.

Reply:
> Using `command-prefix` 

It means we need to add a new input in release e2e template, I do not think it worthy for only a single customized plugin.

> The `nohup` may not work in the windows

I did some test in a windows machine using git bash, it supports `nohup` commands. And for github CI image `windows-latest`, it is using bash as well. 

> close dummy server for other tests

Yeah I take this case into consideration so I take setup work before all of the test cases get triggered in support/index.js.